onnxruntime/core/optimizer/conv_add_act_fusion.cc

```diff
@@ -46,7 +46,10 @@ class ConvAddActivation : public NodeSelector {
 
   std::optional<NodesToOptimizeIndices> Select(const GraphViewer& graph_viewer, const Node& node) const override {
     const std::string_view node_ep = node.GetExecutionProviderType();
-    if (node_ep != kCpuExecutionProvider || !HasElementDataType(*node.InputDefs()[0], ONNX_NAMESPACE::TensorProto_DataType_FLOAT)) {
+    if (node_ep != kCpuExecutionProvider ||
+        (!HasElementDataType(*node.InputDefs()[0], ONNX_NAMESPACE::TensorProto_DataType_FLOAT) &&
+         !HasElementDataType(*node.InputDefs()[0], ONNX_NAMESPACE::TensorProto_DataType_FLOAT16) )
```

Comment:
>TensorProto_DataType_FLOAT16 [](http://example.com/codeflow?start=67&length=28)

This is way too simplistic. The thing is CPU EP does not support fp16 all the time. currently it only support fp16 on ARMv8.v2. This logic does not fly on eariler ARMv8, and it does not fly on x64 either. And that's just the current situation, which might change in later revisions.